apps/session-manager/src/controllers/__tests__/sso.spec.ts

```diff
@@ -0,0 +1,135 @@
+import { describe, test, expect, vi, beforeEach } from "vitest";
+
+import * as express from "express";
+
+import * as E from "fp-ts/Either";
+import * as TE from "fp-ts/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+
+import {
+  ResponseErrorInternal,
+  ResponseErrorNotFound,
+  ResponseErrorTooManyRequests,
+  ResponseSuccessJson,
+} from "@pagopa/ts-commons/lib/responses";
+import { DateFromString } from "@pagopa/ts-commons/lib/dates";
+
+import { FnAppAPIClient } from "../../repositories/fn-app-api";
+
+import * as profileService from "../../services/profile";
+import { getUserForFIMS } from "../sso";
+
+import { RedisClientSelectorType } from "../../types/redis";
+
+import {
+  mockedInitializedProfile,
+  mockedUser,
+} from "../../__mocks__/user.mocks";
+import mockRes from "../../__mocks__/response.mocks";
+import mockReq from "../../__mocks__/request.mocks";
+import { FIMSUser } from "../../generated/fims/FIMSUser";
+
+const mockGetProfile = vi.spyOn(profileService, "getProfile");
+mockGetProfile.mockReturnValue(
+  TE.of(ResponseSuccessJson(mockedInitializedProfile)),
+);
+
+beforeEach(() => {
+  vi.clearAllMocks();
+});
+
+describe("SSOController#getUserForFIMS", () => {
+  const res = mockRes() as unknown as express.Response;
+  const req = mockReq() as unknown as express.Request;
+
+  const expectedFIMSUserResponse: FIMSUser = {
+    acr: mockedUser.spid_level,
+    auth_time: mockedUser.created_at,
+    date_of_birth: pipe(
+      mockedUser.date_of_birth,
+      DateFromString.decode,
+      E.getOrElseW(() => {
+        throw Error("Invalid test initialization");
+      }),
+    ),
+    email: mockedInitializedProfile.email,
+    family_name: mockedUser.family_name,
+    fiscal_code: mockedUser.fiscal_code,
+    name: mockedUser.name,
+  };
+
+  const mockedDependencies = {
+    // Repositories are not used, since we mocked the service layer
+    fnAppAPIClient: {} as ReturnType<typeof FnAppAPIClient>,
+    redisClientSelector: {} as RedisClientSelectorType,
+    user: mockedUser,
+    req,
+  };
+
+  test("when the profile contains a validated email, then it returns a FIMS user with email", async () => {
+    await pipe(
+      mockedDependencies,
+      getUserForFIMS,
+      TE.map((response) => response.apply(res)),
+      TE.mapLeft((err) => expect(err).toBeFalsy()),
+    )();
+
+    expect(res.status).toHaveBeenCalledWith(200);
```

Comment:
This is good enough, but we can check the right path in a different way
```suggestion
    const result = await pipe(
      mockedDependencies,
      getUserForFIMS,
      TE.map((response) => response.apply(res)),
    )();

    expect(E.isLeft(result)).toBeFalsy();
    expect(res.status).toHaveBeenCalledWith(200);
```

Reply:
Done in [[#IOPID-1831] refactor based on comments](https://github.com/pagopa/io-auth-n-identity-domain/pull/21/commits/f7172071277aa811bf0b9d7c392ed059f7a54b29)